src/ALCProxy.Communication/ALCClient.cs

```diff
@@ -57,13 +59,14 @@ public void SetUpServer(AssemblyLoadContext alc, string typeName, string assembl
             _server = constructedType.GetConstructor(
                 new Type[] { typeof(Type), typeof(Type[]), typeof(IList<object>), typeof(IList<Type>) })
                 .Invoke(new object[] { objType, genericTypes, serializedConstArgs.ToList(), argTypes });
-            _callMethod = _server.GetType().GetMethod("CallObject");
+            _serverDelegate = Delegate.CreateDelegate(typeof(ServerCall), _server, constructedType.GetMethod("CallObject"));
```

Comment:
You should make the `_serverDelegate` field be of type `ServerCall` and just call the delegate below instead of using `DynamicInvoke`. That way you'll reap all of the performance gains of using delegates instead of just `MethodInfo.Invoke`.